activity/core.go

```diff
@@ -42,38 +42,39 @@ func CloseAll() {
 	journalCache.Flush()
 }
 
-// Dir gets the directory where journal entries are written.
+// Dir gets the directory where journals are stored.
 func Dir() string {
 	return journalDir
 }
 
-// SetDir sets the directory where journal entries are written.
+// SetDir sets the directory where journals are stored.
 func SetDir(dir string) {
 	journalDir = dir
 }
 
-// GetID returns the Journal ID stored in the context.
-func GetID(ctx context.Context) string {
-	return ctx.Value(Key).(string)
+// GetJournal returns the Journal stored in the context.
+func GetJournal(ctx context.Context) Journal {
+	return ctx.Value(JournalKey).(Journal)
 }
 
-// Record writes a new entry to the journal identified by the ID at `journal.Key` in
+// Record writes a new entry to the journal identified by the ID at `activity.JournalKey` in
 // the provided context. It also writes to the server logs at the debug level. If no ID
-// is registered, the entry is written to the server logs at the warning level. If the
+// is registered, the entry is written to the server logs at the info level. If the
 // ID is an empty string, it uses the ID 'dead-letter-office'.
 //
 // Record creates a new journal for ID if needed, then appends the message to that journal.
 // Records are journaled in the user's cache directory under `wash/activity/ID.log`.
 func Record(ctx context.Context, msg string, a ...interface{}) {
```

Comment:
I'd say the logic in https://github.com/MikaelSmith/wash/blob/history/activity/core.go#L83-L107 should be moved to a `Journal#Record` method. Otherwise, it's weird having an `activity.Journal` type but no way for that type to access its own file without having to go through `activity.Record`.

Reply:
`journalCache` should also be `journalFileCache` or something similar, because it's not caching `Journal` objects but journal files.

Reply:
Could make the dead-letter-office its own Journal via an internal package variable